ethsigner/src/main/java/tech/pegasys/ethsigner/requesthandler/sendtransaction/TransactionTransmitter.java

```diff
@@ -0,0 +1,162 @@
+    // account.
+    final String signedTransactionHexString;
+    try {
+      final RawTransaction rawTransaction = context.getRawTransactionBuilder().build();
+      signedTransactionHexString = serialiser.serialise(rawTransaction);
+    } catch (final IllegalArgumentException e) {
+      LOG.debug("Failed to encode transaction: {}", context.getRawTransactionBuilder(), e);
+      return new JsonRpcBody(JsonRpcError.INVALID_PARAMS);
+    } catch (final Throwable e) {
+      LOG.debug(
+          "Failed to encode/serialise transaction: {}", context.getRawTransactionBuilder(), e);
+      return new JsonRpcBody(JsonRpcError.INTERNAL_ERROR);
+    }
+
+    final JsonRpcRequest sendRawTransaction =
+        new JsonRpcRequest(
+            JSON_RPC_VERSION, JSON_RPC_METHOD, singletonList(signedTransactionHexString));
+    sendRawTransaction.setId(context.getId());
+
+    try {
+      return new JsonRpcBody(Json.encodeToBuffer(sendRawTransaction));
+    } catch (final IllegalArgumentException e) {
+      LOG.debug("JSON Serialisation failed for: {}", sendRawTransaction, e);
+      return new JsonRpcBody(JsonRpcError.INTERNAL_ERROR);
+    }
+  }
+
+  private void sendTransaction(final Buffer bodyContent) {
+    final HttpServerRequest httpServerRequest = context.getInitialRequest();
+    final HttpClientRequest proxyRequest =
+        ethNodeClient.request(
+            httpServerRequest.method(), httpServerRequest.uri(), this::handleResponse);
+
+    proxyRequest.headers().setAll(httpServerRequest.headers());
+    proxyRequest.headers().remove("Content-Length"); // created during 'end'.
+    proxyRequest.setChunked(false);
+    proxyRequest.end(bodyContent);
+  }
+
+  private void handleResponse(final HttpClientResponse response) {
+    logResponse(response);
+    response.bodyHandler(
+        body -> {
+          logResponseBody(body);
+          handleResponseBody(response, body);
+        });
+  }
+
+  private void handleResponseBody(final HttpClientResponse response, final Buffer body) {
+    try {
+      if (response.statusCode() != HttpResponseStatus.OK.code()
+          && retryMechanism.mustRetry(response, body)) {
+        send();
+        return;
+      }
+    } catch (final IOException e) {
+      LOG.info("Retry mechanism failed, reporting error.");
+      reportError(JsonRpcError.INTERNAL_ERROR);
+      return;
+    }
+
+    final HttpServerRequest httpServerRequest = context.getInitialRequest();
+    httpServerRequest.response().setStatusCode(response.statusCode());
+    httpServerRequest.response().headers().setAll(response.headers());
+    httpServerRequest.response().setChunked(false);
+    httpServerRequest.response().end(body);
+  }
+
+  private void logResponse(final HttpClientResponse response) {
+    LOG.debug("Response status: {}", response.statusCode());
+  }
+
+  private void logResponseBody(final Buffer body) {
+    LOG.debug("Response body: {}", body);
+  }
+
+  protected void reportError(final JsonRpcError errorCode) {
```

Comment:
think this could be a private method. also looks like the errorCode is always INTERNAL_ERROR

Reply:
Done. and Done.